homeassistant/components/pushover/notify.py

```diff
@@ -53,6 +54,40 @@ def send_message(self, message='', **kwargs):
 
         data['title'] = kwargs.get(ATTR_TITLE, ATTR_TITLE_DEFAULT)
 
+        # Check for attachment.
+        if ATTR_ATTACHMENT in data:
+            # If attachment is a URL, use requests to open it as a stream.
+            if data[ATTR_ATTACHMENT].startswith('http'):
+                try:
+                    import requests
+                    response = requests.get(
+                        data[ATTR_ATTACHMENT],
+                        stream=True,
+                        timeout=5)
+                    if response.status_code == 200:
+                        # Replace the attachment identifier with file object.
+                        data[ATTR_ATTACHMENT] = response.content
+                except requests.exceptions.RequestException as ex_val:
+                    _LOGGER.error(str(ex_val))
+                    # Remove attachment key to try sending without attachment
+                    del data[ATTR_ATTACHMENT]
+            else:
+                # Not a URL, check valid path first
+                if self._hass.config.is_allowed_path(data[ATTR_ATTACHMENT]):
+                    # try to open it as a normal file.
+                    try:
+                        file_handle = open(data[ATTR_ATTACHMENT], 'rb')
+                        # Replace the attachment identifier with file object.
+                        data[ATTR_ATTACHMENT] = file_handle
+                    except IOError as ex_val:
+                        _LOGGER.error(str(ex_val))
```

Comment:
We don't need to copy the exception argument to string. The argument has a `__str__` magic method.

Reply:
Fixed.